youtube_dl/extractor/camwithher.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class CamWithHerIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?camwithher\.tv/view_video\.php\?viewkey=(?P<id>\w+).*'
+
+    _TESTS = [
+        {
+            'url': 'http://camwithher.tv/view_video.php?viewkey=6e9a24e2c0e842e1f177&page=&viewtype=&category=',
+            'info_dict': {
+                'id': '5644',
+                'ext': 'mp4',
+                'title': 'Periscope Tease',
+            },
+            'params': {
+                'skip_download': True,
+            }
+        },
+        {
+            'url': 'http://camwithher.tv/view_video.php?viewkey=6dfd8b7c97531a459937',
+            'only_matching': True,
+        },
+        {
+            'url': 'http://camwithher.tv/view_video.php?viewkey=b6c3b5bea9515d1a1fc4&page=&viewtype=&category=mv',
+            'info_dict': {
+                'id': '758',
+                'ext': 'flv',
+                'title': 'Gisele in the Bath',
+            },
+            'params': {
+                'skip_download': True,
+            }
```

Comment:
It does not check the other path since you neither download it here nor test the `url` not to contain `mp4:`. Thus you can't be sure whether this test involved the other path or not.